arch/arm/src/sama5/sam_flexcom_spi.c

```diff
@@ -0,0 +1,2116 @@
+    {
+      spiinfo("%08" PRIx32 "->%08" PRIx32 "\n", address, value);
+    }
+#endif
+
+  return value;
+}
+
+/****************************************************************************
+ * Name: flex_spi_putreg
+ *
+ * Description:
+ *  Write a value to an SPI register
+ *
+ ****************************************************************************/
+
+static inline void flex_spi_putreg(struct sam_flex_spidev_s *flex_spi,
+                                   uint32_t value, unsigned int offset)
+{
+  uint32_t address = flex_spi->base + offset;
+
+#ifdef CONFIG_SAMA5_SPI_REGDEBUG
+  if (flex_spi_checkreg(flex_spi, true, value, address))
+    {
+      spiinfo("%08" PRIx32 "<-%08" PRIx32 "\n", address, value);
+    }
+#endif
+
+  putreg32(value, address);
+}
+
+/****************************************************************************
+ * Name: flex_spi_dumpregs
+ *
+ * Description:
+ *   Dump the contents of all SPI registers
+ *
+ * Input Parameters:
+ *   spi - The SPI controller to dump
+ *   msg - Message to print before the register data
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_DEBUG_SPI_INFO
+static void flex_spi_dumpregs(struct sam_flex_spidev_s *flex_spi,
+                              const char *msg)
+{
+  spiinfo("%s:\n", msg);
+  spiinfo("    MR:%08x   SR:%08x  IMR:%08x\n",
+          getreg32(flex_spi->base + SAM_FLEXCOM_SPI_MR_OFFSET),
+          getreg32(flex_spi->base + SAM_FLEXCOM_SPI_SR_OFFSET),
+          getreg32(flex_spi->base + SAM_FLEXCOM_SPI_IMR_OFFSET));
+  spiinfo("  CSR0:%08x CSR1:%08x \n",
+          getreg32(flex_spi->base + SAM_FLEXCOM_SPI_CSR0_OFFSET),
+          getreg32(flex_spi->base + SAM_FLEXCOM_SPI_CSR1_OFFSET));
+  spiinfo("  WPCR:%08x WPSR:%08x\n",
+          getreg32(flex_spi->base + SAM_FLEXCOM_SPI_WPCR_OFFSET),
+          getreg32(flex_spi->base + SAM_FLEXCOM_SPI_WPSR_OFFSET));
+}
+#endif
+
+/****************************************************************************
+ * Name: flex_spi_dev
+ *
+ * Description:
+ *    Given a chip select instance, return a pointer to the parent SPI
+ *    controller instance.
+ *
+ ****************************************************************************/
+
+static inline struct sam_flex_spidev_s *flex_spi_dev(struct sam_flex_spics_s
+                                                     *flex_spics)
+{
+#ifdef CONFIG_SAMA5_FLEXCOM0_SPI
+  if (flex_spics->flex_spino == 0)
+      return &g_flexcom0dev;
+#endif
+#ifdef CONFIG_SAMA5_FLEXCOM1_SPI
+  if (flex_spics->flex_spino == 1)
+      return &g_flexcom1dev;
+#endif
+#ifdef CONFIG_SAMA5_FLEXCOM2_SPI
+  if (flex_spics->flex_spino == 2)
+      return &g_flexcom2dev;
+#endif
+#ifdef CONFIG_SAMA5_FLEXCOM3_SPI
+  if (flex_spics->flex_spino == 3)
+      return &g_flexcom3dev;
+#endif
+#ifdef CONFIG_SAMA5_FLEXCOM4_SPI
+  if (flex_spics->flex_spino == 4)
+      return &g_flexcom4dev;
+#endif 
```

Comment:
maybe better convert to switch-case?

Reply:
I followed the method in sam_spi.c (which is just for SPI0 and SPI1), and changed/extended it. We need to make sure the test isn't done for any interfaces not actually in use, but I'll check to see if we would ever get to this code if not and revise as appropriate.

Reply:
OK - I've misunderstoof what sam_spi.c code was doing here so this is actually wrong. I'll demote this PR back to being a draft while I resolve it.

A good but unintentional find - thanks :)

Reply:
no, I didn't misunderstand but the comments in sam_spi.c were misleading.

@pkarashchenko - I'm not sure a switch method is any neater, unless I misunderstood?


```suggestion
  switch (flex_spics->spino)
    {
#ifdef CONFIG_SAMA5_FLEXCOM0_SPI
     case 0:
      return &g_flexcom0dev;
      break;
 #endif
 #ifdef CONFIG_SAMA5_FLEXCOM1_SPI
      case 1:
        return &g_flexcom1dev;
        break;
#endif
etc
```